zha/application/platforms/light/__init__.py

```diff
@@ -1397,7 +1398,7 @@ def restore_external_state_attributes(
         xy_color: tuple[float, float] | None,
         hs_color: tuple[float, float] | None,
         color_mode: ColorMode | None,
-        effect: str | None,
+        effect: str,
```

Comment:
Same here, this should be reverted IMO.
```suggestion
        effect: str | None,
```